From bb42c732ed74efd8c943d6665d1dc6e530d64caa Mon Sep 17 00:00:00 2001
From: ravindranath <ravindranathx.doddi@intel.com>
Date: Fri, 8 Feb 2013 18:55:21 -0500
Subject: Add libusb support

libusb support is required for userspace usb driver in Bluedroid stack.
Build is fixed for android in io.c by including macro TIMESPEC_TO_TIMEVAL.
This patch adds the Android.mk file for Android, and also switches OS
from Darwin to Linux.

Category: Device Enablement
Domain: USB
Upstream Candidate: No

For: AXIA-1459
Change-Id: I7d3480b9e7aee45e4be0d82d7e65d3b914a615d8
Signed-off-by: Ravindranath Doddi <ravindranathx.doddi@intel.com>
---
 config.h          |   14 ++++++--------
 libusb/Android.mk |   19 +++++++++++++++++++
 libusb/io.c       |    5 +++++
 3 files changed, 30 insertions(+), 8 deletions(-)
 create mode 100644 libusb/Android.mk

diff --git a/config.h b/config.h
index 1b548fb..1689a79 100644
--- a/config.h
+++ b/config.h
@@ -1,6 +1,4 @@
-/* config.h.  Generated from config.h.in by configure.  */
-/* config.h.in.  Generated from configure.ac by autoheader.  */
-/* saintlou: File is identical with PPC and Intel, 10/26/2010 */
+/* config.h. is modified to support bluedroid */
 
 /* Default visibility */
 #define API_EXPORTED __attribute__((visibility("default")))
@@ -18,7 +16,7 @@
 #define HAVE_INTTYPES_H 1
 
 /* Define to 1 if you have the `rt' library (-lrt). */
-/* #undef HAVE_LIBRT */
+#define HAVE_LIBRT 1
 
 /* Define to 1 if you have the <memory.h> header file. */
 #define HAVE_MEMORY_H 1
@@ -52,10 +50,10 @@
 /* #undef NO_MINUS_C_MINUS_O */
 
 /* Darwin backend */
-#define OS_DARWIN /**/
+/* #undef OS_DARWIN */
 
 /* Linux backend */
-/* #undef OS_LINUX */
+#define OS_LINUX /**/
 
 /* Name of package */
 #define PACKAGE "libusb"
@@ -82,10 +80,10 @@
 #define STDC_HEADERS 1
 
 /* Backend handles timeout */
-#define USBI_OS_HANDLES_TIMEOUT /**/
+#undef USBI_OS_HANDLES_TIMEOUT
 
 /* timerfd headers available */
-/* #undef USBI_TIMERFD_AVAILABLE */
+#undef USBI_TIMERFD_AVAILABLE
 
 /* Version number of package */
 #define VERSION "1.0.8"
diff --git a/libusb/Android.mk b/libusb/Android.mk
new file mode 100644
index 0000000..39bb23c
--- /dev/null
+++ b/libusb/Android.mk
@@ -0,0 +1,19 @@
+LOCAL_PATH:= $(call my-dir)
+
+include $(CLEAR_VARS)
+
+LOCAL_SRC_FILES:= \
+ core.c \
+ descriptor.c \
+ io.c \
+ sync.c \
+ os/linux_usbfs.c
+
+LOCAL_C_INCLUDES += \
+ external/libusb/ \
+ external/libusb/libusb/ \
+ external/libusb/libusb/os
+
+LOCAL_MODULE_TAGS:= optional
+LOCAL_MODULE:= libusb
+include $(BUILD_SHARED_LIBRARY)
diff --git a/libusb/io.c b/libusb/io.c
index 1379eb3..132721c 100644
--- a/libusb/io.c
+++ b/libusb/io.c
@@ -36,6 +36,11 @@
 
 #include "libusbi.h"
 
+#define TIMESPEC_TO_TIMEVAL(tv, ts)                                     \
+        do {                                                            \
+                (tv)->tv_sec = (ts)->tv_sec;                            \
+                (tv)->tv_usec = (ts)->tv_nsec / 1000;                   \
+        } while (0)
 /**
  * \page io Synchronous and asynchronous device I/O
  *
-- 
1.7.4.1

